### PR TITLE
New: Shareable configs (fixes #2415)

### LINKED ISSUE #2415: Shareable configs
We are  starting to see a trend where people wrap ESLint just to be able to distribute their own style guides:

* https://github.com/feross/standard
* https://github.com/OtoAnalytics/womply-eslint
* https://www.npmjs.com/package/sc-eslint

It should not be this hard to share configs with others, and is clearly a sign of a need that is not being fulfilled by ESLint.

Now that we support `extends` in config files, we have the perfect way to support shareable configs.

So what I'd propose is that we extend `extends` such that it can accept an npm package name (`eslint-config-*`), which would allow people to publish their configs as packages that export a config object. 

I think this change would be fairly small but would really open up config sharing.

### PR DESCRIPTION
**Pull request by nzakas**

### CODE REVIEW

File: lib/config.js [hunk 1]
```diff
@@ -45,45 +45,77 @@ var loadedPlugins = Object.create(null);
...
+            }
+
+        } else {
+
+            // it's a package
+            if (filePath.indexOf("eslint-config-") === -1) {
+                filePath = "eslint-config-" + filePath;
+            }
+
+            config = require(filePath);
```
Comment (ilyavolodin) [inline lib/config.js]: If package wasn't installed, should we show a more user-friendly message instead of callstack?

**Comment by ilyavolodin** [comment]
You should also add a note in developer-guide as well about the syntax for config packages. Probably to `working with plugins`, although those are not really plugins, so not sure where.

File: lib/config.js (hunk 1 above)
Comment (nzakas) [inline lib/config.js]: We show a call stack if we can't read a config file, so the behavior is consistent.

**Comment by nzakas** [comment]
Yeah, I just couldn't think of a good place to put it in developer guide. It doesn't really fit anywhere, so probably need a new page.

**Pull request merged**